app/controllers/case_information_controller.rb

```diff
@@ -97,10 +122,75 @@ def nomis_offender_id_from_url
     params.require(:nomis_offender_id)
   end
 
+  def display_address_page
+    @prisoner = prisoner(@last_location.nomis_offender_id)
+    render :new
+  end
+
+  def handle_stage2_rendering
+    @prisoner = prisoner(case_information_params[:nomis_offender_id])
+    render :missing_info
+  end
+
+  def team_identifier
+    Team.find_by(name: params['input-autocomplete'])&.id
```

Comment:
It looks like the value in the autocomplete field comes back to us as a team name, and we're using that identify the team – i.e. we find the team by its name, and convert that into its respective ID.

**This is only safe to do if the team names are unique.** Otherwise, we won't be able to identify which team was selected given a duplicate team name.

But it looks like the name uniqueness constraint on team name **only applies to NPS teams**:
https://github.com/ministryofjustice/offender-management-allocation-manager/blob/fa312de2842c019129848e21d5d142fcb5b42fc0/app/models/team.rb#L8

**There's a chance that we have duplicate CRC team names.**

Whilst we won't need to send out automatic handover emails for CRC offenders, it's important to note that we may be capturing the wrong data as it goes in at this end. So CRC team data might not be reliable if needed in the future.

It looks like a crack that things could fall down and come back to bite us in the future.

Are we okay with knowingly taking this on?

Reply:
There is one more commit which I think I need to add to this branch - I was uncertain about it as I couldn't work out how to implement the 'Remove' button (but I think it can just go anyway) which limits the team to those with email addresses. I guess there's something weird here about CRC cases that we have missed - good spot

Reply:
17f6eb1 seems to have done the trick 👍 now there's no ambiguity about duplicate team names – we simply receive the ID.